wps/core/repository.py

```diff
@@ -75,13 +78,25 @@ def __init__(
         )
         if not self._signing_key.has_private:
             raise KeyError("No private work order signing key found.")
+        self._access = access_check
         self._dao = work_package_dao
 
+    # pylint: disable=too-many-locals
     async def create(
         self, creation_data: WorkPackageCreationData, auth_context: AuthContext
     ) -> WorkPackageCreationResponse:
         """Create a work package and store it in the repository."""
 
+        user_id = auth_context.id
+        dataset_id = creation_data.dataset_id
+        work_type = creation_data.type
+
+        if (
```

Comment:
Maybe also raise not implemented error when work type is upload.

Reply:
Done in c63d896.